lib/jeckle/rest_actions.rb

```diff
@@ -13,11 +13,11 @@ def find(id)
       end
 
       def search(params = {})
-        response   = run_request(resource_name, params).response.body || []
+        response   = run_request(params.delete[:resource_name] || resource_name, params).response.body || []
```

Comment:
I'm not sure if it is the best solution, I would like to hear @tomas-stefano and @mateusg.

Regarding this approach I would like to suggest to extract this default handling:

```ruby
def search(params = {})
  custom_resource_name = params.delete[:resource_name] || resource_name

  # ...
end
```

Hummm, thinking about this issue... Do you know that we can redefine `resource_name` in any model that includes `Jeckle::Resource`, and that will be automatically used by `search` method (for example)? Example:

```ruby
class Shot
  include Jeckle::Resource

  api :dribbble

  # Documentation: https://github.com/tomas-stefano/jeckle/blob/master/lib/jeckle/http.rb#L26-L36
  def self.resource_name
    'photos.json'
  end
```